src/NServiceBus.NHibernate/Deduplication/GatewayDeduplication.cs

```diff
@@ -46,12 +46,18 @@ await session.SaveAsync(gatewayMessage)
                     await tx.CommitAsync()
                         .ConfigureAwait(false);
                 }
-                catch (GenericADOException)
+                catch (ConstraintViolationException)
                 {
                     await tx.RollbackAsync()
                         .ConfigureAwait(false);
                     return false;
                 }
+                catch (GenericADOException)
+                {
+                    await tx.RollbackAsync()
+                        .ConfigureAwait(false);
+                    throw;
+                }
```

Comment:
@DavidBoike yeah. I guess so. The other option would be to now catch it at all as I believe that calling `Rollback` explicitly is optional here because a not committed session rolls back.

Reply:
Took off `Generic` and all the tests passed. :smile: